js/modals/question-modal.js

```diff
@@ -2,74 +2,74 @@ $(function() {
+      if (!$button.hasClass('js-active')) {
+         var $activeButton = $('.modal .js-hint.js-hint-active');
+         var $hintsTextarea = $('.modal .js-hints-textarea');
+
+         $activeButton
+             .data('hint-text', $hintsTextarea.val())
+             .removeClass('js-hint-active')
+             .find('span')
+             .text($activeButton.data('hint'));
 
-         $('fieldset.hints .hint.active').data('hint-text', $('fieldset.hints textarea').val());
-         $('fieldset.hints textarea').val($button.data('hint-text'));
-         $('fieldset.hints textarea').attr('placeholder', 'Hint ' + $button.data('hint'));
-         $('.hint.active>span').text($('.hint.active').data('hint'));
-         $('.hint.active').removeClass('active');
+         $hintsTextarea
+            .val($button.data('hint-text'))
+            .attr('placeholder', 'Hint ' + $button.data('hint'));
 
          $button
-             .addClass('active')
+             .addClass('js-hint-active')
              .find('span')
              .text('Hint ' + $button.data('hint'));
 
-         $('fieldset.hints textarea').focus();
+         $hintsTextarea.focus();
       }
    }
 
    function createHints() {
-      var $hints = $('<fieldset class="form-group hints"> <label>Hints </label></fieldset>');
-      var hintButton1 = '<button type="button" class="btn btn-default hint hint-1 active" data-hint="1"><span>Hint 1</span></button>';
-      var hintButton2 = '<button type="button" class="btn btn-default hint hint-2" data-hint="2"><span>2</span></button>';
-      var hintButton3 = '<button type="button" class="btn btn-default hint hint-3" data-hint="3"><span>3</span></button>';
-      var $hintTextarea = $('<textarea class="form-control" id="q-statement-input" rows="3" placeholder="Hint 1"></textarea>');
-
-      $hints
-          .append(hintButton1)
-          .append(hintButton2)
-          .append(hintButton3)
-          .append($hintTextarea);
-
-      $hintTextarea.on('keyup', function() {
+      var $templates = $('.qa-templates');
+      var hints = $templates.find('.js-modal-question .js-hints')
+          .clone()
+          .children();
+
+      $('.modal').on('keyup', '.js-hints-textarea', function() {
          $('.hint.active').data('hint-text', $(this).val());
       });
 
-      $('.modal').on('click', '.hint', function() {
+      $('.modal').on('click', '.js-hint', function() {
          hintButtonClick($(this));
       });
 
-      return $hints;
+      return hints;
    }
 
    function createMultipleChoiceOptions(n) {
-      var $r = $('');
+      var $options = $('');
+      var $templates = $('.qa-templates');
 
       for(var i = 0; i < n; i++) {
          var input = '<input type="text" class="form-control wrong" id="q-mc-option-' + i + '" placeholder="Option ' +
-             (i + 1 + $('.mc-text-option').length) + ' Text">';
+             (i + 1 + $('.multiple-choice>.mc-text-option').length) + ' Text">';
+
```

Comment:
Have `$('.multiple-choice>.mc-text-option').length` outside of the loop and assign it into a variable so that you don't have to recalculate it at every iteration.